generators/cleanup.js

```diff
@@ -16,3 +17,13 @@ function cleanupOldFiles(generator, javaDir, testDir) {
         generator.removeFile(ANGULAR_DIR + 'components/form/uib-pagination.config.js');
     }
 }
+
+/**
+ * Removes server files that where generated in previous JHipster versions and therefore need to be removed
+ */
+function cleanupOldServerFiles(generator, javaDir, testDir) {
+    if (generator.isJhipsterVersionLessThan('3.4.3')) {
+        generator.removeFile(javaDir + 'domain/util/JSR310DateTimeSerializer.java');
```

Comment:
I think it would be better to use ```generator.fs.delete()``` but there is currently an issue (https://github.com/yeoman/generator/issues/950) so we can see about that later.